scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -658,39 +652,119 @@ def test_as_euler_degenerate_symmetric_axes():
         [15, 0, 60],
         [35, 0, 75],
         [60, 180, 35],
-        [15, -180, 25],
-        ])
+        [15, -180, 25]])
 
     with pytest.warns(UserWarning, match="Gimbal lock"):
-        for axis1 in ['x', 'y', 'z']:
-            for axis2 in ['x', 'y', 'z']:
-                if axis1 == axis2:
-                    continue
+        for seq_tuple in permutations('xyz'):
+            # Extrinsic rotations
+            seq = ''.join([seq_tuple[0], seq_tuple[1], seq_tuple[0]])
+            rotation = Rotation.from_euler(seq, angles, degrees=True)
+            mat_expected = rotation.as_matrix()
 
-                # Extrinsic rotations
-                seq = axis1 + axis2 + axis1
-                rotation = Rotation.from_euler(seq, angles, degrees=True)
-                mat_expected = rotation.as_matrix()
+            angle_estimates = rotation.as_euler(seq, degrees=True)
+            mat_estimated = Rotation.from_euler(
+                seq, angle_estimates, degrees=True
+                ).as_matrix()
+
+            assert_array_almost_equal(mat_expected, mat_estimated)
+
+            # Intrinsic rotations
+            seq = seq.upper()
+            rotation = Rotation.from_euler(seq, angles, degrees=True)
+            mat_expected = rotation.as_matrix()
 
-                angle_estimates = rotation.as_euler(seq, degrees=True)
-                mat_estimated = Rotation.from_euler(
-                    seq, angle_estimates, degrees=True
-                    ).as_matrix()
+            angle_estimates = rotation.as_euler(seq, degrees=True)
+            mat_estimated = Rotation.from_euler(
+                seq, angle_estimates, degrees=True
+                ).as_matrix()
 
-                assert_array_almost_equal(mat_expected, mat_estimated)
+            assert_array_almost_equal(mat_expected, mat_estimated)
+
+
+def test_as_euler_compare_algorithms():
+    rnd = np.random.RandomState(0)
+    n = 10
```

Comment:
It can be easily increased to 1000 or even 10000.

Reply:
Done!